src/Compilers/CSharp/Test/Emit3/OverloadResolutionPriorityTests.cs

```diff
@@ -0,0 +1,1997 @@
+        };
+
+        CreateCompilation([executable, source, OverloadResolutionPriorityAttributeDefinition]).VerifyDiagnostics(expectedDiagnostics);
+
+        var comp = CreateCompilation([source, OverloadResolutionPriorityAttributeDefinition]);
+        CreateCompilation(executable, references: [AsReference(comp, useMetadataReference)]).VerifyDiagnostics(expectedDiagnostics);
+    }
+
+    [Theory, CombinatorialData]
+    public void UnaryOperator_Allowed(bool useMetadataReference)
+    {
+        var source = """
+            using System.Runtime.CompilerServices;
+
+            public struct S
+            {
+                public static S? operator!(S? s) => throw null;
+
+                [OverloadResolutionPriority(1)]
+                public static S operator!(S s)
+                {
+                    System.Console.Write("1");
+                    return s;
+                }
+            }
+            """;
+
+        var executable = """
+            S? s = new S();
+            _ = !s;
+            """;
+
+        CompileAndVerify([executable, source, OverloadResolutionPriorityAttributeDefinition], expectedOutput: "1").VerifyDiagnostics();
+
+        var comp = CreateCompilation([source, OverloadResolutionPriorityAttributeDefinition]);
+        CompileAndVerify(executable, references: [AsReference(comp, useMetadataReference)], expectedOutput: "1").VerifyDiagnostics();
+    }
+
+    [Theory]
+    [InlineData("implicit")]
+    [InlineData("explicit")]
+    public void ConversionOperators_Disallowed(string operatorType)
+    {
+        var source = $$"""
+            using System.Runtime.CompilerServices;
+
+            public class C
+            {
+                [OverloadResolutionPriority(1)]
+                public static {{operatorType}} operator C(int i) => throw null;
+            }
+            """;
+
+        CreateCompilation([source, OverloadResolutionPriorityAttributeDefinition]).VerifyDiagnostics(
+            // (5,6): error CS9501: Cannot use 'OverloadResolutionPriorityAttribute' on this member.
+            //     [OverloadResolutionPriority(1)]
+            Diagnostic(ErrorCode.ERR_CannotApplyOverloadResolutionPriorityToMember, "OverloadResolutionPriority(1)").WithLocation(5, 6)
+        );
+    }
+
+    [Fact]
+    public void DisallowedOnStaticCtors()
+    {
+        var code = """
+            using System.Runtime.CompilerServices;
+
+            public class C
+            {
+                [OverloadResolutionPriority(1)]
+                static C() => throw null;
+            }
+            """;
+
+        CreateCompilation([code, OverloadResolutionPriorityAttributeDefinition]).VerifyDiagnostics(
+            // (5,6): error CS9501: Cannot use 'OverloadResolutionPriorityAttribute' on this member.
+            //     [OverloadResolutionPriority(1)]
+            Diagnostic(ErrorCode.ERR_CannotApplyOverloadResolutionPriorityToMember, "OverloadResolutionPriority(1)").WithLocation(5, 6)
+        );
+    }
+}
```

Comment:
Have we tried applying the attribute twice?

Reply:
From test plan review, some other suggestions:
- expression tree. Expression<> x = (a) => A.B(a); // priority should be honored, silent source break like params? can yield an error identify and document known scenario, notify BCL team (apply with care)
- interpolation handlers
- query syntax